base/server/src/com/netscape/cms/profile/def/SANToCNDefault.java

```diff
@@ -0,0 +1,125 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package com.netscape.cms.profile.def;
+
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.util.Locale;
+
+import org.mozilla.jss.netscape.security.x509.CertificateSubjectName;
+import org.mozilla.jss.netscape.security.x509.DNSName;
+import org.mozilla.jss.netscape.security.x509.GeneralName;
+import org.mozilla.jss.netscape.security.x509.GeneralNameInterface;
+import org.mozilla.jss.netscape.security.x509.GeneralNames;
+import org.mozilla.jss.netscape.security.x509.PKIXExtensions;
+import org.mozilla.jss.netscape.security.x509.SubjectAlternativeNameExtension;
+import org.mozilla.jss.netscape.security.x509.X500Name;
+import org.mozilla.jss.netscape.security.x509.X509CertInfo;
+
+import com.netscape.certsrv.profile.EProfileException;
+import com.netscape.certsrv.property.IDescriptor;
+import com.netscape.certsrv.request.IRequest;
+
+/**
+ * This plug-in replaces the subject DN with CN=<DNS name>
+ * using the first DNS name in the SAN extension.
```

Comment:
Do we want to use the first SAN extension? I'd kinda guess we'd want to use a wildcard if it exists, else the one closest to the root (`.`) (though tie breaking between multiple domains by first-seen would probably be fine.

Reply:
Technically does it matter which DNS name we pick to become the CN? Otherwise, while it's certainly nice to have, probably it can be done later when time permits. The current code doesn't support wildcard yet, so we need to fix that first. Then we need to come up with an algorithm to rank the DNS names so we can select one to become the CN. And if the CSR already has a CN that ranks lower than the other DNS names in SAN, do we replace the CN?

Reply:
I don't think it matters.  Do the simplest thing.

Reply:
ACK, this is fine then.